cmake/modules/RootBuildOptions.cmake

```diff
@@ -152,6 +152,7 @@ ROOT_BUILD_OPTION(rpath OFF "Set run-time library load path on executables and s
 ROOT_BUILD_OPTION(ruby OFF "Ruby ROOT bindings, requires ruby >= 1.8")
 ROOT_BUILD_OPTION(runtime_cxxmodules OFF "Enable runtime support for C++ modules.")
 ROOT_BUILD_OPTION(sapdb OFF "MaxDB/SapDB support, requires libsqlod and libsqlrte")
+ROOT_BUILD_OPTION(simdzlib OFF "SIMD version of zlib, requires builtin_zlib")
```

Comment:
Is it not possible to enable this in some other way than adding yet another build option to ROOT? Worst case we could check that the hardware supports this and enable by default if `builtin_zlib=ON`. What do you think?

Reply:
Looking at the code it seems that the tests for SSE and AVX/AVX2 are performed at runtime, so it might be possible to always enable this by default.

Reply:
@amadio I inroduced ```simzlib``` option to separate two options: to have pure update to zlib 1.2.11 in ``builtin_zlib``` (before it was 1.2.8) and to introduce SIMD acceleration on demand only for x86_64 systems with Linux...I had an idea to have hardware checks, but somehow I was recommended to try such option. 

Unfortunately, I spend some time on this PR, and ```simdzlib``` is not only protect intrinsic based code, but also code with function__attribute(s)__ that are not supported by some compilers and some architectures, like ARM for example.